tests/test_pagination.py

```diff
@@ -152,16 +150,42 @@ def pagination_test(self):
                                     msg="A paginated synced stream has a record that is missing automatic fields.")
 
                     # Verify we have more fields sent to the target than just automatic fields (this is set above)
-                    self.assertEqual(auto_fields.difference(actual_keys),
-                                     set(), msg="A paginated synced stream has a record that is missing expected fields.")
+                    self.assertEqual(set(), auto_fields.difference(actual_keys),
+                                     msg="A paginated synced stream has a record that is missing expected fields.")
 
-                # TODO ADD CHECK ON IDS
-                # Verify by pks that the data replicated matches what we expect
+                primary_keys = self.expected_primary_keys().get(stream)
+                pks = list(primary_keys) if primary_keys else None
+                if not pks:
+                    pks = self.makeshift_primary_keys().get(stream)
 
+                # Verify by pks that the replicated records match our expectations
+                self.assertRecordsEqualByPK(stream, expected_records.get(stream), actual_records, pks)
+
+                # Verify the expected number of records were replicated
+                self.assertEqual(len(expected_records.get(stream)), len(actual_records))
+
+
+    def assertRecordsEqualByPK(self, stream, expected_records, actual_records, pks):
```

Comment:
Feels like we could reuse this in multiple tests and pull this into the base method